Add mulesoft to java compatibility docs (#10154)
DataDog/documentation: content/en/tracing/setup_overview/compatibility_requirements/java.md

| Java Servlet Compatible | 2.3+, 3.0+ | Fully Supported | `servlet`, `servlet-2`, `servlet-3`            |
| Jax-RS Annotations      | JSR311-API | Fully Supported | `jax-rs`, `jaxrs`, `jax-rs-annotations`, `jax-rs-filter` |
| Jetty                   | 7.0-9.x    | Fully Supported | `jetty`                                        |

> | Mulesoft                | 4          | Fully Supported | `mule`                                         |

Reviewer: Mule is not enabled by default, and is dependent on the Grizzly server and client instrumentations that are marked as `[Beta][2]`

**Note**: Many application servers are Servlet compatible and are automatically covered by that instrumentation, such as Tomcat, Jetty, Websphere, Weblogic, and JBoss.
Also, frameworks like Spring Boot inherently work because it usually uses a supported embedded application server (Tomcat/Jetty/Netty).

> The Mulesoft Instrumentation is off by default, to enable add the following settings:

Reviewer 2: ```suggestion
Mulesoft Instrumentation is not enabled by default. To enable, add the following settings:
```

**Note**: Many application servers are Servlet compatible and are automatically covered by that instrumentation, such as Tomcat, Jetty, Websphere, Weblogic, and JBoss.
Also, frameworks like Spring Boot inherently work because it usually uses a supported embedded application server (Tomcat/Jetty/Netty).

**Integrations Disabled By Default**

> The following instrumentation is off by default and can be enabled with the following settings:

Reviewer: ```suggestion
The following instrumentations are disabled by default and can be enabled with the following settings:
```

- Postgres SQL
- ScalikeJDBC

**Integrations Disabled By Default**

> The following instrumentation is off by default and can be enabled with the following settings:

Reviewer: ```suggestion
The following instrumentations are disabled by default and can be enabled with the following settings:
```
